File: Minyoung06/PGS_12977.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        boolean[] isPrime = new boolean[3001]; // 최악의 경우 1000+1000+1000
        int N = isPrime.length;
        
        Arrays.fill(isPrime,true);
        isPrime[0] = isPrime[1] = false;
        
        for(int i=2; i<=Math.sqrt(N); i++){
            for(int j=i*i; j<N; j+=i){
                isPrime[j]=false;
            }
        }
        
        int len = nums.length;
        for(int i=0; i<len-2; i++){
            for(int j=i+1; j<len-1; j++){
                for(int k=j+1; k<len; k++){
                    if(isPrime[nums[i]+nums[j]+nums[k]]) 
                        answer++;
                }
            }
        }
        
        return answer;
    }
}

File: Minyoung06/PGS_12977.cpp
#include <vector>
#include <iostream>
using namespace std;

int solution(vector<int> nums) {
    int N = 3001; // 최댓값; 1000+1000+1000
    vector<bool> isPrime(N,true);
    isPrime[0] = isPrime[1] = false;
    
    //에라토스테네스의 체
    for(int i=2; i*i<N; i++){
        if(isPrime[i]){
            for(int j=i*i; j<=N; j+=i){
                isPrime[j] = false;
            }
        }
    }
    
    int answer = 0;
    int len = nums.size();
    for(int i=0; i<len-2; i++){
        for(int j=i+1; j<len-1; j++){
            for(int k=j+1; k<len; k++){
                if (isPrime[nums[i]+nums[j]+nums[k]]) answer++;
            }
        }
    }
    return answer;
}